Allow plugin developers to use custom label property.

sebastianlenz opened the pull request:
Without this fix it is not possible to use a custom labelProperty for config.xml values. With this fix you can use e.g.: <labelProperty>displayName</labelProperty> in config.xml for entity salutation.

example value for <component> in config.xml

<component name="sw-entity-single-select">
    <name>salutation</name>
    <entity>salutation</entity>
    <labelProperty>displayName</labelProperty>
    <label>Salutation</label>
    <label lang="de-DE">Anrede</label>
</component>

<!--
Thank you for contributing to Shopware! Please fill out this description template to help us to process your pull request.

Please make sure to fulfil our contribution guideline (https://docs.shopware.com/en/shopware-platform-dev-en/community/contribution-guideline?category=shopware-platform-dev-en/community).

Do your changes need to be mentioned in the documentation?
Add notes on your change right now in the documentation files in /src/Docs/Resources and add them to the pull request as well. 
-->

### 1. Why is this change necessary?


### 2. What does this change do, exactly?


### 3. Describe each step to reproduce the issue or behaviour.


### 4. Please link to the relevant issues (if any).


### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfil them.

janbuecker commented:
Thanks for your PR!

Unfortunately, we won't implement this new `xtype` in near future, because it is not absolutely necessary at the moment.

This ticket will be solved by changing the misleading labels.

sebastianlenz committed:
Allow plugin developers to use custom label property.

Without this fix it is not possible to use a custom labelProperty for config.xml values. With this fix you can use e.g.: <labelProperty>displayName</labelProperty> in config.xml for entity salutation.

example value for <component> in config.xml

<component name="sw-entity-single-select">
    <name>salutation</name>
    <entity>salutation</entity>
    <labelProperty>displayName</labelProperty>
    <label>Salutation</label>
    <label lang="de-DE">Anrede</label>
</component>

sebastianlenz commented:
This fix is not neccessary. It works even without this change. Please add this to documentation.
